presto-main/src/main/java/com/facebook/presto/server/ResourceGroupStateInfoResource.java

```diff
@@ -76,6 +103,9 @@ public ResourceGroupStateInfoResource(
         this.resourceGroupManager = requireNonNull(resourceGroupManager, "resourceGroupManager is null");
         this.internalNodeManager = requireNonNull(internalNodeManager, "internalNodeManager is null");
         this.proxyHelper = requireNonNull(proxyHelper, "proxyHelper is null");
+        this.resourceGroupStateInfoKeySupplierMap = new HashMap<>();
+        this.expirationDuration = requireNonNull(serverConfig, "serverConfig is null").getResourceGroupStateInfoExpirationDuration();
+        this.rootResourceGroupInfoSupplier = expirationDuration.getValue() > 0 ? memoizeWithExpiration(() -> resourceGroupManager.getRootResourceGroups(), expirationDuration.toMillis(), MILLISECONDS) : Suppliers.ofInstance(resourceGroupManager.getRootResourceGroups());
```

Comment:
why do we need `Suppliers.ofInstance` ? 
can we just do` () -> resourceGroupManager.getRootResourceGroups()`